### PR TITLE
fix MOAB options for v5.0

### PR DESCRIPTION
**Pull request by gonuke**

### CODE REVIEW

**Comment by makeclean** [comment]
THis looks great, I'll wait for CI to finish for the free daily test

File: doc/install/dependencies.rst [hunk 1]
```diff
@@ -82,8 +82,7 @@ is required.
 ::
```
Comment (ljacobson64) [inline doc/install/dependencies.rst:81]: Can we remove the note just above this saying that `--enable-dagmc` is required?

File: doc/install/dependencies.rst [hunk 2]
```diff
@@ -98,8 +97,7 @@ configure option is required.
...
```
Comment (ljacobson64) [inline doc/install/dependencies.rst:95]: Here too

**Comment by makeclean** [comment]
that option is removed now

**Comment by ljacobson64** [comment]
I know, which is why we shouldn't tell people it's required anymore

**Comment by makeclean** [comment]
Oh I see, sorry, I read that as can we remove the note above the enable-dagmc part :~{

**Comment by pshriwise** [comment]
so is this PR in a "good enough" state for merging. This an other PR's are beginning to get stale...

**Comment by ljacobson64** [comment]
Just submitted a PR to Paul's branch that makes the changes I requested.

**Comment by ljacobson64** [comment]
Thanks for merging Paul. This is fine to merge after the CI passes imo

**Pull request merged**